src/report/markdown.rs

```diff
@@ -0,0 +1,137 @@
+            let prev_total_coverable: usize = previous_data
+                .iter()
+                .map(|(path, _)| previous_data.coverable_in_path(path))
+                .sum();
+
+            // Calculate the coverage percentage for the previous result
+            let prev_coverage_percentage = if prev_total_coverable > 0 {
+                (prev_total_covered as f64 / prev_total_coverable as f64) * 100.0
+            } else {
+                0.0
+            };
+
+            let diff = coverage_percentage - prev_coverage_percentage;
+            if diff.abs() < 0.01 {
+                format!(
+                    "{}/{} ({:.2}%)",
+                    total_covered, total_coverable, coverage_percentage
+                )
+            } else {
+                let sign = if diff > 0.0 { "+" } else { "-" };
+                format!(
+                    "{}/{} ({:.2}% {}{:.2}%)",
+                    total_covered,
+                    total_coverable,
+                    coverage_percentage,
+                    sign,
+                    diff.abs()
+                )
+            }
+        }
+        None => format!(
+            "{}/{} ({:.2}%)",
+            total_covered, total_coverable, coverage_percentage
+        ),
+    };
+
+    // Add the base coverage info to the markdown
+    let mut markdown_content = format!(
+        "# Coverage Report\n\nOverall coverage: {}\n\n",
+        coverage_info
+    );
+
+    // Next, add file-by-file coverage table
+    markdown_content.push_str("## File Coverage\n\n");
+    markdown_content.push_str("| File | Coverage | Percentage | Change |\n");
+    markdown_content.push_str("|------|----------|------------|--------|\n");
+
+    // Loop through each file and add its coverage info to the markdown
+    for (path, _) in coverage_data.iter() {
+        let covered = coverage_data.covered_in_path(path);
+        let coverable = coverage_data.coverable_in_path(path);
+        let file_percentage = if coverable > 0 {
+            (covered as f64 / coverable as f64) * 100.0
+        } else {
+            0.0
+        };
+
+        let change_info = match previous_result {
+            Some(ref previous_data) => {
+                let prev_covered = previous_data.covered_in_path(path);
+                let prev_coverable = previous_data.coverable_in_path(path);
+                let prev_file_percentage = if prev_coverable > 0 {
+                    (prev_covered as f64 / prev_coverable as f64) * 100.0
+                } else {
+                    0.0
+                };
+
+                let diff = file_percentage - prev_file_percentage;
+                if diff.abs() < 0.01 {
+                    "-".to_string()
+                } else {
+                    let sign = if diff > 0.0 { "+" } else { "-" };
+                    format!("{}{:.2}%", sign, diff.abs())
+                }
+            }
+            None => "-".to_string(),
+        };
+
+        let path_str = path.to_string_lossy();
```

Comment:
So one consideration with this is that it's the absolute path of the file. You might want to mention the root directory somewhere else and then use a relative path instead (just to avoid one potentially very wide column)

Reply:
Rock on, done and done. Unsure if there's a better way to calculate that relative path, open to alternative more-cleanlier options